crates/bevy_log/src/android_tracing.rs

```diff
@@ -85,13 +88,15 @@ impl<S: Subscriber + for<'a> LookupSpan<'a>> Layer<S> for AndroidLayer {
             Level::WARN => android_log_sys::LogPriority::WARN,
             Level::ERROR => android_log_sys::LogPriority::ERROR,
         };
-        let message = format!("{}\0", recorder);
-        let tag = format!("{}\0", meta.name());
+        let message = CString::new(recorder).expect("message contained null character.");
+        let tag = CString::new(meta.name()).expect("tag contained null character.");
```

Comment:
Would it be better to filter out any null char before creating the `CString`? That way the log behavior would match the rust side if the passed strings contain any null byte (they would just be skipped when printing).

Reply:
I agree with this eventual approach, any null bytes should be converted to [the replacement character](https://doc.rust-lang.org/stable/std/primitive.char.html#associatedconstant.REPLACEMENT_CHARACTER). This might need benching, though.

I think this PR should be merged as-is any probably marked for 0.13.2, then we can create a follow-up PR that removes panicking.

Reply:
Done. The allocation needs iteration anyway. This should be identical other than using memcpy vs byte-by-byte checks here. This will be slower for big strings, but this shouldn't be a performance critical path in many apps.